lib/vquic/quiche.c

```diff
@@ -297,6 +297,10 @@ CURLcode Curl_quic_is_connected(struct connectdata *conn, int sockindex,
   }
 
   return result;
+  error:
+  quiche_config_free(qs->cfg);
+  quiche_conn_free(qs->conn);
+  return result;
```

Comment:
This does not seem to be the right place for such checks. I don't see any earlier place here that allocates these two resources. Thinking aloud (not sure if this is all feasible!):

Based on a quick look, the call chain is:
```
multi_runsingle in lib/multi.c (as one of the users)
-> Curl_is_connected
  -> Curl_quic_is_connected
```

In `multi_runsingle`, it seems to invoke the `multi_done` function which seems to call `conn->handler->done` if available. Not sure if that is a good point to hook into, but it might be an option.

There is also the socket creator itself in lib/connect.c, `Curl_quic_connect`. On error it calls `Curl_closesocket`. Maybe somewhere down that chain, this could also be cleaned up?

Reply:
> I don't see any earlier place here that allocates these two resources. 

If they weren't allocated, then I'm sure they wouldn't leak! =) They're allocated in `Curl_quic_connect`.

> On error it calls Curl_closesocket. Maybe somewhere down that chain, this could also be cleaned up?

That's easier for TCP and UDP when the data is all under the socket. For QUIC, that's not the case and it seemed easier to not add any more internal quic function APIs since the connecting function can clean it up by itself.

Reply:
> If they weren't allocated, then I'm sure they wouldn't leak! =) They're allocated in `Curl_quic_connect`.

Ha, what I meant was that it isn't clear in this function that the memory was allocated. :)